src/alasqlparser.jison

```diff
@@ -2018,29 +2020,48 @@ PrimaryKey
 
 ForeignKey
 	: FOREIGN KEY LPAR ColsList RPAR REFERENCES Table ParColsList?
-	     OnForeignKeyClause
-		{ $$ = {type: 'FOREIGN KEY', columns: $4, fktable: $7, fkcolumns: $8}; }
+	     OnReferentialActions
+		{ $$ = {type: 'FOREIGN KEY', columns: $4, fktable: $7, fkcolumns: $8}; yy.extend($$, $9); }
 	;
 
 ParColsList
 	: LPAR ColsList RPAR
 		{ $$ = $2; }
 	;
 
-OnForeignKeyClause
+OnReferentialActions
 	:
-		{ $$ = undefined; }
+		{$$ = {}; }
+	| OnDeleteClause
+		{$$ = {ondelete: $1}; }
+	| OnUpdateClause
+		{$$ = {onupdate: $1}; }
 	| OnDeleteClause OnUpdateClause
-		{ $$ = undefined; }
+		{$$ = {ondelete: $1, onupdate: $2}; }
+	| OnUpdateClause OnDeleteClause
+		{$$ = {ondelete: $2, onupdate: $1}; }
 	;
 
 OnDeleteClause
-	: ON DELETE NO ACTION
-		{$$ = undefined; }
+	: ON DELETE ReferentialAction
+		{$$ = $3; }
 	;
 OnUpdateClause
-	: ON UPDATE NO ACTION
-		{$$ = undefined; }
```

Comment:
Please confirm if we have existing tests covering on delete no action and on update no action - and the aspects we are touching with  FOREIGN KEY. 

If we dont, please make sure we add tests to cover a comprehensive set of edge cases and combinations that can reveal if we broke functionality with this PR. 

The first goal is to remain as SQL-99 compliant as before. The second goal is to remain backwards compatibility.